hubblestack/extmods/hubble_mods/readfile.py

```diff
@@ -0,0 +1,512 @@
+             each member of the value list if ``valsep`` is defined) and turn the
+             result into a key-value pair in a dictionary. If this is defined in
+             conjunction with ``valsep``, the result will be a dictionary, not
+             a list of single-key dictionaries.
+
+    chaining (If used in chaining)
+             Chained values will be called with ``.format()`` on the ``path``.
+
+- For file format: string
+    path - Path of file
+    encode_64 - (Optional) Set to `True` if the return string should be base64 encoded.
+        Defaults to `False` which returns a regular string.
+
+    chaining (If used in chaining)
+        Chained values will be called with ``.format()`` on the ``path``.
+"""
+
+import os
+import logging
+import json as _json
+import yaml as _yaml
+import re
+
+from hubblestack.utils.encoding import encode_base64
+import hubblestack.extmods.module_runner.runner_utils as runner_utils
+from hubblestack.utils.hubble_error import HubbleCheckValidationError
+
+log = logging.getLogger(__name__)
+
+def validate_params(block_id, block_dict, extra_args=None):
+    """
+    Validate all mandatory params required for this module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': '/some/path', 'status': True},
+                  'caller': 'Audit'}
+
+    Raises:
+        AuditCheckValidationError: For any validation error
+    """
+    log.debug('Module: readfile Start validating params for check-id: {0}'.format(block_id))
+
+    filepath = runner_utils.get_param_for_module(block_id, block_dict, 'path')
+    file_format = runner_utils.get_param_for_module(block_id, block_dict, 'format')
+
+    error = {}
+    if not filepath:
+        error['path'] = 'No filepath provided'
+    if not file_format:
+        error['format'] = 'No file format provided'
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+def execute(block_id, block_dict, extra_args=None):
+    """
+    Execute the module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': '/some/path', 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing readfile module for id: {0}'.format(block_id))
+
+    chain_args = None if not extra_args else extra_args.get('chaining_args')
+    file_format = runner_utils.get_param_for_module(block_id, block_dict, 'format')
+    if file_format == 'json':
+        return _handle_file(file_format, block_id, block_dict, chain_args)
+    elif file_format == 'yaml':
```

Comment:
no need of separate else condition here. can be combined with if clause

Reply:
It was just for separation. I actually combined the code later on.
Resolved for now.